common/upload.js

```diff
@@ -376,12 +403,40 @@
 
                     // Once all checksums have been calculated call createUploadJobs
                     // To create a job for each file
-                    if (vm.checksumCompleted == vm.noOfFiles) {
-                        createUploadJobs();
+                    if (vm.checksumCompleted == vm.filesCt) {
+                        preCheckFileExists();
                     }
                 }
             };
 
+            uploadFile.prototype.onPreFileExistSuccess = function() {
+
+                if (vm.erred || vm.aborted) return;
+
+                this.skipUploadJob = this.hatracObj.jobDone;
+
+                // if the job is already done, that means the file has an idntical file already in the server
+                // md5, filename, and size all match
```

Comment:
Since this check is in ermrestjs and I think we might want to change it (removing check for the filename), let's just say identical file and remove the actual details of what is the definition of identical file. (we don't need to mention we're checking for md5, filename, etc.)